core/services/llo/data_source.go

```diff
@@ -103,27 +109,34 @@ func (d *dataSource) Observe(ctx context.Context, streamValues llo.StreamValues,
 			stream, exists := d.registry.Get(streamID)
 			if !exists {
 				errmu.Lock()
-				errors = append(errors, ErrObservationFailed{streamID: streamID, reason: fmt.Sprintf("missing stream: %d", streamID)})
+				errs = append(errs, ErrObservationFailed{streamID: streamID, reason: fmt.Sprintf("missing stream: %d", streamID)})
 				errmu.Unlock()
 				promMissingStreamCount.WithLabelValues(fmt.Sprintf("%d", streamID)).Inc()
 				return
 			}
 			run, trrs, err := stream.Run(ctx)
 			if err != nil {
 				errmu.Lock()
-				errors = append(errors, ErrObservationFailed{inner: err, run: run, streamID: streamID, reason: "pipeline run failed"})
+				errs = append(errs, ErrObservationFailed{inner: err, run: run, streamID: streamID, reason: "pipeline run failed"})
 				errmu.Unlock()
 				promObservationErrorCount.WithLabelValues(fmt.Sprintf("%d", streamID)).Inc()
+				// TODO: Consolidate/reduce telemetry. We should send all observation results in a single packet
+				// https://smartcontract-it.atlassian.net/browse/MERC-6290
+				d.t.EnqueueV3PremiumLegacy(run, trrs, streamID, opts, nil, err)
 				return
 			}
+			// TODO: Consolidate/reduce telemetry. We should send all observation results in a single packet
+			// https://smartcontract-it.atlassian.net/browse/MERC-6290
 			val, err = ExtractStreamValue(trrs)
 			if err != nil {
 				errmu.Lock()
-				errors = append(errors, ErrObservationFailed{inner: err, run: run, streamID: streamID, reason: "failed to extract big.Int"})
+				errs = append(errs, ErrObservationFailed{inner: err, run: run, streamID: streamID, reason: "failed to extract big.Int"})
 				errmu.Unlock()
 				return
 			}
 
+			d.t.EnqueueV3PremiumLegacy(run, trrs, streamID, opts, val, nil)
```

Comment:
We want to monitor streams not channels since channels can contain multiple streams, which does not map properly to the telemetry packets.

This is just one of those impedance mismatches. We are trying to force a channel/stream paradigm into the old feed telemetry structure.

The real solution is to change the telemetry packets and report solely stream values, forgetting feeds entirely, but we can't do that without breaking compatibility right now.